api/__tests__/routes/user_region.test.ts

```diff
@@ -0,0 +1,404 @@
+                            zip_code: 9000,
+                        },
+                        address_id: 2,
+                        admin: false,
+                        date_added: "2023-05-04T12:00:00.000Z",
+                        deleted: false,
+                        email: "superstudent@trottoir.be",
+                        first_name: "Toon",
+                        id: 2,
+                        last_login: "2023-05-04T12:00:00.000Z",
+                        last_name: "De Superstudent",
+                        phone: "9876543210",
+                        student: false,
+                        super_student: true,
+                    },
+                    user_id: 2,
+                },
+            ];
+            await runner.get({
+                url: "/user_region",
+                expectedData: expected,
+            });
+        });
+
+        afterAll(() => {
+            app.close();
+        });
+    });
+
+    describe("Unsuccesful requests", () => {
+        let runner: Testrunner;
+        beforeAll(async () => {
+            const server = request(app);
+            runner = new Testrunner(server);
+
+            await deleteDatabaseData();
+            await initialiseDatabase();
+        });
+
+        describe("Must be correctly authorized to use any path", () => {
+            const new_user_region = {
+                user_id: 1,
+                region_id: 3,
+            };
+
+            describe("Cannot reach any path without authorisation", () => {
+                beforeEach(() => {
+                    runner.authLevel(AuthenticationLevel.UNAUTHORIZED);
+                });
+
+                test("Cannot reach GET /user_region", async () => {
+                    await runner.get({
+                        url: "/user_region",
+                        expectedData: [forbiddenResponse],
+                        statusCode: 403,
+                    });
+                });
+
+                test("Cannot reach GET /user_region/:id", async () => {
+                    await runner.get({
+                        url: "/user_region/1",
+                        expectedData: [forbiddenResponse],
+                        statusCode: 403,
+                    });
+                });
+
+                test("Cannot reach POST /user_region", async () => {
+                    await runner.post({
+                        url: "/user_region",
+                        data: new_user_region,
+                        expectedResponse: forbiddenResponse,
+                        statusCode: 403,
+                    });
+                });
+
+                test("Cannot reach DELETE /user_region/:id", async () => {
+                    await runner.delete({
+                        url: "/user_region/1",
+                        statusCode: 403,
+                    });
+                });
+            });
+            describe("Cannot reach any path as a student", () => {
```

Comment:
Er ontbreken tests voor `updateOne` en `getOne`.

Reply:
een student kan momenteel voor elke id getOne doen. Ik heb daar een issue voor aangemaakt #355. Ik zal in deze issue de testen aanvullen voor getOne